Source/Core/DolphinQt/Config/FileChooserWidget.h

```diff
@@ -0,0 +1,36 @@
+// Copyright 2015 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <memory>
+#include <QWidget>
+
+// Predefinitions
+namespace Ui
+{
+class DFileChooser;
+}
+
+class DFileChooser final : public QWidget
+{
+	Q_OBJECT
+
+public:
+	explicit DFileChooser(QWidget* parent_widget = nullptr);
+	~DFileChooser();
+
+	inline void setFormat(bool folder, QString filter) { m_folder = folder; m_filter = filter; }
```

Comment:
Functions defined in the class definition are already inline by default.

Taking QString by reference would be nicer, IMO (I know Qt uses CoW, but const PBR indicates you don't expect modifications in the signature without having to see the code).